iocore/net/P_SNIActionPerformer.h

```diff
@@ -341,15 +343,25 @@ class TLSValidProtocols : public ActionItem
 #endif
   TLSValidProtocols() : protocol_mask(max_mask) {}
   TLSValidProtocols(unsigned long protocols) : unset(false), protocol_mask(protocols) {}
+  TLSValidProtocols(int min_ver, int max_ver) : unset(false), protocol_mask(0), min_ver(min_ver), max_ver(max_ver) {}
 
   int
   SNIAction(TLSSNISupport *snis, const Context & /* ctx */) const override
   {
-    if (!unset) {
-      auto ssl_vc            = dynamic_cast<SSLNetVConnection *>(snis);
+    if (this->min_ver >= 0 || this->max_ver >= 0) {
       const char *servername = snis->get_sni_server_name();
-      Debug("ssl_sni", "TLSValidProtocol param 0%x, fqdn [%s]", static_cast<unsigned int>(this->protocol_mask), servername);
-      ssl_vc->set_valid_tls_protocols(protocol_mask, TLSValidProtocols::max_mask);
+      Debug("ssl_sni", "TLSValidProtocol min=%d, max=%d, fqdn [%s]", this->min_ver, this->max_ver, servername);
+      auto ssl_vc = dynamic_cast<SSLNetVConnection *>(snis);
+      ssl_vc->set_valid_tls_version_min(this->min_ver);
+      ssl_vc->set_valid_tls_version_max(this->max_ver);
+    } else {
```

Comment:
Why not `} else if (!unset) {` ?

Reply:
I guess I wanted to make it clear that `if (!unset) {` is the first line for the case min/max isn't used. It enables us to add some code before the `if` easily.